integration/benchmark/account/keys.go

```diff
@@ -0,0 +1,108 @@
+package account
+
+import (
+	"fmt"
+	"sync"
+
+	flowsdk "github.com/onflow/flow-go-sdk"
+
+	"github.com/onflow/flow-go-sdk/crypto"
+)
+
+var ErrNoKeysAvailable = fmt.Errorf("no keys available")
+
+type accountKey struct {
+	*flowsdk.AccountKey
+
+	mu      sync.Mutex
+	ks      *keystore
+	Address *flowsdk.Address
+	Signer  crypto.InMemorySigner
+	inuse   bool
+}
+
+type keystore struct {
+	availableKeys chan *accountKey
+	size          int
+}
+
+func newKeystore(keys []*accountKey) *keystore {
+	ks := &keystore{}
+
+	availableKeys := make(chan *accountKey, len(keys))
+	for _, key := range keys {
+		key.ks = ks
+		availableKeys <- key
+	}
+	ks.size = len(keys)
+	ks.availableKeys = availableKeys
+
+	return ks
+}
+
+func (k *keystore) Size() int {
+	return k.size
+}
+
+func (k *keystore) getKey() (*accountKey, error) {
+	select {
+	case key := <-k.availableKeys:
+		key.mu.Lock()
+		defer key.mu.Unlock()
+
+		if key.inuse {
+			return nil, fmt.Errorf("key already in use")
+		}
+		key.inuse = true
+
+		return key, nil
+	default:
+		return nil, ErrNoKeysAvailable
+	}
+}
+
+func (k *accountKey) markUnused() {
+	k.mu.Lock()
+	defer k.mu.Unlock()
+
+	k.inuse = false
+}
+
+// Done unlocks a key after use and puts it back into the pool.
+func (k *accountKey) Done() {
+	k.markUnused()
+	k.ks.availableKeys <- k
+}
+
+// Success is called when a key was successfully used to sign a transaction as the proposer
+// It increments the sequence number.
+func (k *accountKey) Success() {
```

Comment:
success is a little misleading since it also needs to be called if the tx failed. Maybe rename to `Increment()`?

Reply:
Made is a bit more verbose to explicitly saw what's incremented